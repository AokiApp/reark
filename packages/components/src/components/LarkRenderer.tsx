import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
