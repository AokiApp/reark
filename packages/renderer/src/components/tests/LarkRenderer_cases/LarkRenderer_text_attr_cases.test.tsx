import { renderWithVRT } from "../test-utils/renderWithVRT";
import { LarkRenderer } from "../../LarkRenderer";

// LarkRenderer 属性バリエーション複合
describe("LarkRenderer 属性バリエーション複合", () => {
  it("textブロック: 太字・斜体・下線・打消し・インラインコード・コメント付き", async () => {
    const textBlock = {
      block_id: "text-attr-1",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          { text_run: { content: "太字", text_element_style: { bold: true } } },
          {
            text_run: { content: "斜体", text_element_style: { italic: true } },
          },
          {
            text_run: {
              content: "下線",
              text_element_style: { underline: true },
            },
          },
          {
            text_run: {
              content: "打消し",
              text_element_style: { strikethrough: true },
            },
          },
          {
            text_run: {
              content: "インラインコード",
              text_element_style: { inline_code: true },
            },
          },
          {
            text_run: {
              content: "コメント付き",
              text_element_style: { comment_ids: ["c1"] },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };

    const blocks = [textBlock];

    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 組み合わせ違い
describe("LarkRenderer 属性バリエーション: 組み合わせ違い", () => {
  it("textブロック: 太字＋下線、斜体＋打消し、インラインコード＋コメント付き", async () => {
    const block = {
      block_id: "attr-combo",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "太字＋下線",
              text_element_style: { bold: true, underline: true },
            },
          },
          {
            text_run: {
              content: "斜体＋打消し",
              text_element_style: { italic: true, strikethrough: true },
            },
          },
          {
            text_run: {
              content: "インラインコード＋コメント付き",
              text_element_style: { inline_code: true, comment_ids: ["c2"] },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 太字のみ
describe("LarkRenderer 属性バリエーション: 太字のみ", () => {
  it("textブロック: 太字のみ", async () => {
    const block = {
      block_id: "bold-only",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "太字のみ",
              text_element_style: { bold: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 斜体のみ
describe("LarkRenderer 属性バリエーション: 斜体のみ", () => {
  it("textブロック: 斜体のみ", async () => {
    const block = {
      block_id: "italic-only",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "斜体のみ",
              text_element_style: { italic: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 下線のみ
describe("LarkRenderer 属性バリエーション: 下線のみ", () => {
  it("textブロック: 下線のみ", async () => {
    const block = {
      block_id: "underline-only",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "下線のみ",
              text_element_style: { underline: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 打消しのみ
describe("LarkRenderer 属性バリエーション: 打消しのみ", () => {
  it("textブロック: 打消しのみ", async () => {
    const block = {
      block_id: "strike-only",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "打消しのみ",
              text_element_style: { strikethrough: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: インラインコードのみ
describe("LarkRenderer 属性バリエーション: インラインコードのみ", () => {
  it("textブロック: インラインコードのみ", async () => {
    const block = {
      block_id: "inlinecode-only",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "インラインコードのみ",
              text_element_style: { inline_code: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: コメント付きのみ
describe("LarkRenderer 属性バリエーション: コメント付きのみ", () => {
  it("textブロック: コメント付きのみ", async () => {
    const block = {
      block_id: "comment-only",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "コメント付きのみ",
              text_element_style: { comment_ids: ["c3"] },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 太字＋斜体
describe("LarkRenderer 属性バリエーション: 太字＋斜体", () => {
  it("textブロック: 太字＋斜体", async () => {
    const block = {
      block_id: "bold-italic",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "太字＋斜体",
              text_element_style: { bold: true, italic: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: 下線＋打消し
describe("LarkRenderer 属性バリエーション: 下線＋打消し", () => {
  it("textブロック: 下線＋打消し", async () => {
    const block = {
      block_id: "underline-strike",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "下線＋打消し",
              text_element_style: { underline: true, strikethrough: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: インラインコード＋太字
describe("LarkRenderer 属性バリエーション: インラインコード＋太字", () => {
  it("textブロック: インラインコード＋太字", async () => {
    const block = {
      block_id: "inlinecode-bold",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "インラインコード＋太字",
              text_element_style: { inline_code: true, bold: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});

// LarkRenderer 属性バリエーション: コメント＋斜体
describe("LarkRenderer 属性バリエーション: コメント＋斜体", () => {
  it("textブロック: コメント＋斜体", async () => {
    const block = {
      block_id: "comment-italic",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "コメント＋斜体",
              text_element_style: { comment_ids: ["c4"], italic: true },
            },
          },
        ],
        style: { align: 1 as const, folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});
