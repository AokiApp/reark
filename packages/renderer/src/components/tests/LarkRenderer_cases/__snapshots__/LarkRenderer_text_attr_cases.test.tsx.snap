// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 属性バリエーション: インラインコードのみ textブロック: インラインコードのみ 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--inline-code reark-text"
      >
        インラインコードのみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: インラインコードのみ textブロック: インラインコードのみ 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--inline-code reark-text"
      >
        インラインコードのみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: インラインコード＋太字 textブロック: インラインコード＋太字 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--inline-code reark-text"
      >
        インラインコード＋太字
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: インラインコード＋太字 textブロック: インラインコード＋太字 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--inline-code reark-text"
      >
        インラインコード＋太字
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: コメント付きのみ textブロック: コメント付きのみ 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <div
        class="reark-comment"
        data-comment-ids="c3"
      >
        <span
          class="reark-text"
        >
          コメント付きのみ
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: コメント付きのみ textブロック: コメント付きのみ 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <div
        class="reark-comment"
        data-comment-ids="c3"
      >
        <span
          class="reark-text"
        >
          コメント付きのみ
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: コメント＋斜体 textブロック: コメント＋斜体 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <div
        class="reark-comment"
        data-comment-ids="c4"
      >
        <span
          class="reark-text--italic reark-text"
        >
          コメント＋斜体
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: コメント＋斜体 textブロック: コメント＋斜体 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <div
        class="reark-comment"
        data-comment-ids="c4"
      >
        <span
          class="reark-text--italic reark-text"
        >
          コメント＋斜体
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 下線のみ textブロック: 下線のみ 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--underline reark-text"
      >
        下線のみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 下線のみ textブロック: 下線のみ 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--underline reark-text"
      >
        下線のみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 下線＋打消し textブロック: 下線＋打消し 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--strikethrough reark-text--underline reark-text"
      >
        下線＋打消し
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 下線＋打消し textブロック: 下線＋打消し 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--strikethrough reark-text--underline reark-text"
      >
        下線＋打消し
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 太字のみ textブロック: 太字のみ 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text"
      >
        太字のみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 太字のみ textブロック: 太字のみ 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text"
      >
        太字のみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 太字＋斜体 textブロック: 太字＋斜体 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--italic reark-text"
      >
        太字＋斜体
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 太字＋斜体 textブロック: 太字＋斜体 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--italic reark-text"
      >
        太字＋斜体
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 打消しのみ textブロック: 打消しのみ 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--strikethrough reark-text"
      >
        打消しのみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 打消しのみ textブロック: 打消しのみ 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--strikethrough reark-text"
      >
        打消しのみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 斜体のみ textブロック: 斜体のみ 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--italic reark-text"
      >
        斜体のみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 斜体のみ textブロック: 斜体のみ 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--italic reark-text"
      >
        斜体のみ
      </span>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 組み合わせ違い textブロック: 太字＋下線、斜体＋打消し、インラインコード＋コメント付き 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--underline reark-text"
      >
        太字＋下線
      </span>
      <span
        class="reark-text--italic reark-text--strikethrough reark-text"
      >
        斜体＋打消し
      </span>
      <div
        class="reark-comment"
        data-comment-ids="c2"
      >
        <span
          class="reark-text--inline-code reark-text"
        >
          インラインコード＋コメント付き
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション: 組み合わせ違い textブロック: 太字＋下線、斜体＋打消し、インラインコード＋コメント付き 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--underline reark-text"
      >
        太字＋下線
      </span>
      <span
        class="reark-text--italic reark-text--strikethrough reark-text"
      >
        斜体＋打消し
      </span>
      <div
        class="reark-comment"
        data-comment-ids="c2"
      >
        <span
          class="reark-text--inline-code reark-text"
        >
          インラインコード＋コメント付き
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション複合 textブロック: 太字・斜体・下線・打消し・インラインコード・コメント付き 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text"
      >
        太字
      </span>
      <span
        class="reark-text--italic reark-text"
      >
        斜体
      </span>
      <span
        class="reark-text--underline reark-text"
      >
        下線
      </span>
      <span
        class="reark-text--strikethrough reark-text"
      >
        打消し
      </span>
      <span
        class="reark-text--inline-code reark-text"
      >
        インラインコード
      </span>
      <div
        class="reark-comment"
        data-comment-ids="c1"
      >
        <span
          class="reark-text"
        >
          コメント付き
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`LarkRenderer 属性バリエーション複合 textブロック: 太字・斜体・下線・打消し・インラインコード・コメント付き 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text"
      >
        太字
      </span>
      <span
        class="reark-text--italic reark-text"
      >
        斜体
      </span>
      <span
        class="reark-text--underline reark-text"
      >
        下線
      </span>
      <span
        class="reark-text--strikethrough reark-text"
      >
        打消し
      </span>
      <span
        class="reark-text--inline-code reark-text"
      >
        インラインコード
      </span>
      <div
        class="reark-comment"
        data-comment-ids="c1"
      >
        <span
          class="reark-text"
        >
          コメント付き
        </span>
      </div>
    </div>
  </div>
</div>
`;
